wadler_lindig/_definitions.py

```diff
@@ -275,17 +275,30 @@ def _pformat_function(obj: types.FunctionType, **kwargs) -> AbstractDoc:
 
 def _pformat_dataclass(obj, **kwargs) -> AbstractDoc:
     type_name = "_" + type(obj).__name__
-    objs = named_objs(
-        [
-            (
-                field.name.removeprefix(type_name),
-                getattr(obj, field.name, _WithRepr("<uninitialised>")),
-            )
-            for field in dataclasses.fields(obj)
-            if field.repr and field.default is dataclasses.MISSING
-        ],
-        **kwargs,
-    )
+    if kwargs["show_defaults"] is True:
```

Comment:
Just `if kwargs["show_defaults"]` will work here, no need to compare booleans to booleans to get another boolean out ;)